add option to run local operators

maelk opened the pull request:
Instead of using 'make deploy' for Baremetal Operator and CAPI
provider baremetal, add an option to use 'make run' and log the
output.

maelk committed:
add option to run local operators

Instead of using 'make deploy' for Baremetal Operator and CAPI
provider baremetal, add an option to use 'make run' and log the
output.

nordixinfra commented:
Can one of the admins verify this patch?

kanwar-saad commented on 03_launch_mgmt_cluster.sh:
```diff
@@ -7,6 +7,7 @@ source lib/logging.sh
 source lib/common.sh
 
 eval "$(go env)"
+export GOPATH
```
a small header for the script would be nice .. now i dont understand in the first glance that what are the inputs for this script.

kanwar-saad commented on 03_launch_mgmt_cluster.sh:
```diff
@@ -63,7 +67,15 @@ function launch_minikube() {
+      touch bmo.out.log
+      touch bmo.err.log
+      make deploy
+      kubectl scale deployment metal3-baremetal-operator -n metal3 --replicas=0
```
can we just delete those pods or deployment?

kanwar-saad commented on 04_verify.sh:
```diff
@@ -105,9 +105,12 @@ check_k8s_entity() {
+      FAILS="$(equals "$(echo "${ENTITY}" | jq -r '.status.readyReplicas')" \
+        "$(echo "${ENTITY}" | jq -r '.status.replicas')" \
+        "${name} ${1} replicas correct")"
+    fi
```
shouldn't there be an else checking that local process is running?

maelk committed:
add doc and test for running operators locally

- add code documentation for environment variables
- add test to check that the operators are running locally
- add user example
